src/js/controllers/mapController.ts

```diff
@@ -1183,6 +1188,289 @@ export class MapController {
+            const specificLayer = this._map?.findLayerById(layerID);
+            if (specificLayer) {
+              if (specificLayer.id.includes('7d')) {
+                specificLayer.visible = true;
+              } else {
+                specificLayer.visible = false;
+              }
+            }
+          });
+        }
+        break;
+      default:
+        break;
+    }
+  }
+
+  setVIIRSDefinedRange(layer: any, sublayerType: string): void {
+    if (!this._map) {
+      return;
+    }
+    const VIIRS24 = layer.sublayers.items.filter((sublayer: Sublayer) =>
+      sublayer.title.includes('24 hrs')
+    );
+
+    switch (sublayerType) {
+      case '24 hrs':
+        {
+          VIIRS24.visible = true;
+          VIIRSLayerIDs.forEach(({ layerID }) => {
+            const specificLayer = this._map?.findLayerById(layerID);
+
+            if (specificLayer) {
+              specificLayer.visible = false;
+            }
+          });
+        }
+        break;
+      case '48 hrs':
+        {
+          VIIRS24.visible = false;
+          VIIRSLayerIDs.forEach(({ layerID }) => {
+            const specificLayer = this._map?.findLayerById(layerID);
+            if (specificLayer) {
+              if (specificLayer.id.includes('48')) {
+                specificLayer.visible = true;
+              } else {
+                specificLayer.visible = false;
+              }
+            }
+          });
+        }
+        break;
+      case '72 hrs':
+        {
+          VIIRS24.visible = false;
+          VIIRSLayerIDs.forEach(({ layerID }) => {
+            const specificLayer = this._map?.findLayerById(layerID);
+            if (specificLayer) {
+              if (specificLayer.id.includes('72')) {
+                specificLayer.visible = true;
+              } else {
+                specificLayer.visible = false;
+              }
+            }
+          });
+        }
+        break;
+      case '7 days':
+        {
+          VIIRS24.visible = false;
+          VIIRSLayerIDs.forEach(({ layerID }) => {
+            const specificLayer = this._map?.findLayerById(layerID);
+            if (specificLayer) {
+              if (specificLayer.id.includes('7d')) {
+                specificLayer.visible = true;
+              } else {
+                specificLayer.visible = false;
+              }
+            }
+          });
+        }
+        break;
+      default:
+        break;
+    }
+  }
+
+  resetDefinedDateRange(layerID: string): void {
+    if (!this._map) {
+      return;
+    }
+
+    const layer = (this._map.allLayers as any).items.filter(
+      (layer: LayerProps) => layer.id === layerID
+    )[0];
+
+    if (!layer || !layer.sublayers) {
+      return;
+    }
+
+    const sublayer24 = layer.sublayers.items.filter((sublayer: Sublayer) =>
+      sublayer.title.includes('24 hrs')
+    );
+
+    sublayer24.visible = false;
+
+    if (layer.id.includes('VIIRS_ACTIVE_FIRES')) {
```

Comment:
do we expect layer id to be something else here ? I am not sure why are we using `includes` and not a strict equality comparison